src/effects/effectchainslot.h

```diff
@@ -35,6 +35,7 @@ class EffectChainSlot : public QObject {
 
     void loadEffectChain(EffectChainPointer pEffectChain);
     EffectChainPointer getEffectChain() const;
+    EffectChainPointer getAndEnsureEffectChain(EffectsManager* pEffectsManager);
```

Comment:
the function "ensures" that the returned chain is valid. I am happy to switch to a better name. 
Do you have one? "getOrCreate ...?"